src/gfx.cpp

```diff
@@ -123,7 +125,61 @@ bool GFX_t::load()
         m_loadErrors = 0;
     }
 
-    // Add new optional assets here. Also update load_gfx.cpp:loadCustomUIAssets()
+    loadImage(Backdrop, uiPath + "Backdrop.png");
+    loadImage(Backdrop_Border, uiPath + "Backdrop_Border.png");
+
+    IniProcessing ini;
+    ini.open(uiPath + "Backdrop_Border.ini");
+    loadFrameInfo(ini, g_backdropBorderInfo);
+
+    if(Backdrop_Border.inited)
+    {
+        // warn if invalid
+        const FrameBorderInfo& i = g_backdropBorderInfo;
+        if(i.le + i.li + i.ri + i.re > Backdrop_Border.w)
+            pLogWarning("Invalid border: total internal/external width is %d but Backdrop_Border.png is only %dpx wide.", i.le + i.li + i.ri + i.re, Backdrop_Border.w);
+        if(i.te + i.ti + i.bi + i.be > Backdrop_Border.h)
+            pLogWarning("Invalid border: total internal/external height is %d but Backdrop_Border.png is only %dpx wide.", i.te + i.ti + i.bi + i.be, Backdrop_Border.h);
+    }
+
+    if(m_loadErrors > 0)
+    {
+        pLogDebug("Missing new backdrop textures.");
+        m_loadErrors = 0;
+    }
+
+    loadImage(WorldMapFrame_Tile, uiPath + "WorldMapFrame_Tile.png");
+    loadImage(WorldMapFrame_Border, uiPath + "WorldMapFrame_Border.png");
+    loadImage(WorldMapFog, uiPath + "WorldMapFog.png");
+
+    ini.open(uiPath + "WorldMapFrame_Border.ini");
+    loadFrameInfo(ini, g_worldMapFrameBorderInfo);
+
+    if(WorldMapFrame_Border.inited)
+    {
+        // warn if invalid
+        const FrameBorderInfo& i = g_worldMapFrameBorderInfo;
+        if(i.le + i.li + i.ri + i.re > WorldMapFrame_Border.w)
+            pLogWarning("Invalid border: total internal/external width is %d but WorldMapFrame_Border.png is only %dpx wide.", i.le + i.li + i.ri + i.re, WorldMapFrame_Border.w);
+        if(i.te + i.ti + i.bi + i.be > WorldMapFrame_Border.h)
+            pLogWarning("Invalid border: total internal/external height is %d but WorldMapFrame_Border.png is only %dpx wide.", i.te + i.ti + i.bi + i.be, WorldMapFrame_Border.h);
+    }
```

Comment:
Probably reasonable to unload this image as it doesn't meet the required conditions to just don't use it at all. After this warning, the image will still remain loaded and used. Or why do you keep it remaining?

Reply:
It would be reasonable to de-init the image here, yes. This check is also used internally in `gfx_frame.cpp`, so it wouldn't be used regardless, but I could save some CPU by de-initing. The main thing I don't remember is whether it is safe to de-init a custom GFX or not, since we would also want to be able to do that.